### PR TITLE
Better use of AdvancedResponseWriter

### PR DESCRIPTION
**Pull request by yurikus**
## Description

1. Moved ProcessResponseStream() call to the bottom of ExtractResponseData() to make HttpResponse members available to AdvancedResponseWriter.
2. Check for AdvancedResponseWriter first and call only that or fallback to ResponseWriter.

## Purpose
This pull request is a:

- [X] Improvement

### CODE REVIEW

File: RestSharp/Http.cs [hunk 1]
```diff
@@ -430,18 +426,24 @@ private void ExtractResponseData(HttpResponse response, HttpWebResponse webRespo
...
+
                 webResponse.Close();
             }
         }
 
         private void ProcessResponseStream(Stream webResponseStream, HttpResponse response)
         {
-            if (ResponseWriter == null)
-                response.RawBytes = webResponseStream.ReadAsBytes();
+            if (AdvancedResponseWriter != null)
```
Comment (alexeyzimarev) [inline RestSharp/Http.cs]: What is the purpose of this change? For this method, the previous code did exactly the same, didn't it?

Comment (yurikus) [reply RestSharp/Http.cs]: It did not to the same thing. When ResponseWriter is null and AdvancedResponseWriter is non-null, stream will be consumed by ReadAsBytes() call.

**Comment by alexeyzimarev** [comment]
Thank you for the PR! :+1:  In order to merge this, I need to have a unit test that is failing on the existing version and goes green for the PR, so we know what value it brings.

**Comment by PaulVrugt** [comment]
This fix would also help us a lot. Please hurry ;)

**Comment by yurikus** [comment]
I added a test.

**Comment by PaulVrugt** [comment]
so, when are we releasing this? We are anxiously awaiting this fix

**Comment by PaulVrugt** [comment]
@alexeyzimarev can we create a release containing this fix?
